### ISSUE TITLE
Calling an interface that has [DisableDefaultIncluded] removes the "data" part of all relationships not listed in ?include

### ISSUE THREAD

**Issue by Bendilin25** [joukevandermaas/saule#198]
Hi, 
I have an interface that returns a class with at least two relationships and that interface is set to [DisableDefaultIncluded]. 

A normal call (GET http://localhost/processes) returns everything as it should. 
```
{
	"data": [{
		"type": "processes",
		"id": "61e41d54-08d0-4314-8117-1f8b3436d486",
		"links": {
			"self": "http://localhost/processes/61e41d54-08d0-4314-8117-1f8b3436d486/"
		},
		"attributes": {
			"name": "Process112",
			"description": "this is a description"
		},
		"relationships": {
			"owner": {
				"links": {
					"self": "http://localhost/processes/61e41d54-08d0-4314-8117-1f8b3436d486/relationships/owner/",
					"related": "http://localhost/processes/61e41d54-08d0-4314-8117-1f8b3436d486/owner/"
				},
				"data": {
					"type": "accounts",
					"id": "7395f63d-3fea-4b1f-884f-fcf19ce66c7a"
				}
			},
			"lastEditedBy": {
				"links": {
					"self": "http://localhost/processes/61e41d54-08d0-4314-8117-1f8b3436d486/relationships/last-edited-by/",
					"related": "http://localhost/processes/61e41d54-08d0-4314-8117-1f8b3436d486/last-edited-by/"
				},
				"data": {
					"type": "accounts",
					"id": "de758e91-618f-49c7-be22-67dc3eb23272"
				}
			}
		}
	}],
	"links": {
		"self": "http://localhost:80/processes"
	}
} 
```

However calling GET http://localhost/processes?include=owner results in the removal of the "data" attribute for the other relationship that was not in the ?included list, in the below example "lastEditedBy".

```
{
	"data": [{
		"type": "processes",
		"id": "61e41d54-08d0-4314-8117-1f8b3436d486",
		"links": {
			"self": "http://localhost/processes/61e41d54-08d0-4314-8117-1f8b3436d486/"
		},
		"attributes": {
			"name": "Process112",
			"description": "this is a description"
		},
		"relationships": {
			"owner": {
				"links": {
					"self": "http://localhost/processes/61e41d54-08d0-4314-8117-1f8b3436d486/relationships/owner/",
					"related": "http://localhost/processes/61e41d54-08d0-4314-8117-1f8b3436d486/owner/"
				},
				"data": {
					"type": "accounts",
					"id": "7395f63d-3fea-4b1f-884f-fcf19ce66c7a"
				}
			},
			"lastEditedBy": {
				"links": {
					"self": "http://localhost/processes/61e41d54-08d0-4314-8117-1f8b3436d486/relationships/last-edited-by/",
					"related": "http://localhost/processes/61e41d54-08d0-4314-8117-1f8b3436d486/last-edited-by/"
				}
			}
		}
	}],
	"links": {
		"self": "http://localhost:80/processes"
	},
	"included": [{
		"type": "accounts",
		"id": "7395f63d-3fea-4b1f-884f-fcf19ce66c7a",
		"links": {
			"self": "http://localhost/accounts/7395f63d-3fea-4b1f-884f-fcf19ce66c7a/"
		},
		"attributes": {
			"loginName": "user1",
			"firstname": null,
			"lastName": null,
			"email": ""
		}
	}]
}
```

Cheers

**Comment by joukevandermaas** [comment]
I think the first call should not include *any* `data` hash for either relationship; the purpose of this `DisableDefaultIncluded` attribute is to disable the serialization of potentially heavy objects.

That said, I wonder what you mean by "interface". This attribute is only usable on action methods or controllers. If you put it somewhere else, it is ignored. Indeed the behavior you see is intended in the case where there is no `DisableDefaultIncluded` attribute.

What are you trying to do? Perhaps there is some other way to accomplish it.

**Comment by PhyberApex** [comment]
It is used in a controller. The problem is as you can see in the first call in data->relationships->owner there is a data field with the type and id. Same goes for data->relationships->lastEditedBy. This is correct behavious. HOWEVER if you use the include query parameter (owner in the second example) to include the resources you can see in the second example that:
data->relationships->owner still has data with type and id but for
data->relationships->lastEditedBy the complete data field is missing

~Cheers

**Comment by joukevandermaas** [comment]
> You can see in the first example [...] there is a data field with the type and id. [...] This is correct behaviour.

I don't understand why this is correct behavior. The way I understand the spec, when you specify fields to include, the response should *only* include those fields. So no data hash for relationships that are not included.

The attribute is meant to exclude everything by default, meaning that there should be no data hash for *any* relationship when you have the attribute and no include query parameter.

I'm totally open to being convinced, but I cannot find your expectation reflected in the spec.

Either way, my priority for bugs like this is usually the following:

1. works seamlessly with ember data / another somewhat commonly used client (regardless of what the spec says)
2. does what the spec says
3. does what makes sense at the time

So if this is a case of option 1 i'm totally open to changing the behavior. In that case please explain what you are trying to do and why this would fix it.

**Comment by PhyberApex** [comment]
Oh...I see. That would be our bad then sorry. I just re-read the JSONAPi specification and indeed you are right. It does say there only needs to be ONE OF data, links or meta. I thought the data should always be here.

So it would make this a case 3 I think.

The solution would be to either not include the data for relationships in example 1 OR include it in example 2 as well just to be consistent throughout the library to avoid confusion in the future. Should I leave this issue open or close it and open a new one specificly for this?

~Cheers

**Comment by joukevandermaas** [comment]
Please leave this one open. I'll add the bug label because I think the inconsistency is just a bug, not intentional. Thanks!

**Comment by joukevandermaas** [comment]
I think this got fixed in #224.
